openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -2,37 +2,43 @@
 This module contains various configuration settings via
 waffle switches for the Block Structure framework.
 """
+from edx_django_utils.cache import RequestCache
+from edx_toggles.toggles.__future__ import WaffleSwitch
 
-
-from edx_toggles.toggles import LegacyWaffleSwitch, LegacyWaffleSwitchNamespace
 from openedx.core.lib.cache_utils import request_cached
 
 from .models import BlockStructureConfiguration
 
-# Namespace
-WAFFLE_NAMESPACE = u'block_structure'
-
 # Switches
-INVALIDATE_CACHE_ON_PUBLISH = u'invalidate_cache_on_publish'
-STORAGE_BACKING_FOR_CACHE = u'storage_backing_for_cache'
-RAISE_ERROR_WHEN_NOT_FOUND = u'raise_error_when_not_found'
+INVALIDATE_CACHE_ON_PUBLISH = WaffleSwitch(
+    "block_structure.invalidate_cache_on_publish", __name__
+)
+# The value of this waffle switch may be manually overridden in the context of a request,
+# via `enable_storage_backing_for_cache_in_request`. Thus, checking the value of this switch
+# should be performed via the `is_storage_backing_enabled` function.
+STORAGE_BACKING_FOR_CACHE = WaffleSwitch(
```

Comment:
I'm going to wait until tests are passing to do a more detailed review.  Here are a couple of quick thoughts:
1. Can we renamed to `_STORAGE_BACKING_FOR_CACHE` to make it slightly more obvious it should be used outside?  It's definition could even be moved immediately before `is_storage_backing_enabled` to keep it closer to where it should be used.
2. Note sure if it would help to add a `_SWITCH` suffix to these names so that we remove the old names and add new names, rather than changing these from strings to WaffleSwitches.  Something to consider.

Reply:
> Can we renamed to _STORAGE_BACKING_FOR_CACHE to make it slightly more obvious it should be used outside? 

Yes, but the variable is used extensively in unit tests, for overriding. 

> Note sure if it would help to add a _SWITCH suffix to these names so that we remove the old names and add new names, rather than changing these from strings to WaffleSwitches. Something to consider.

I'm not sure either. 

In those cases where I don't have a strong opinion I'd rather abstain from renaming the variables.

Reply:
Feel free to skip the `_SWITCH` name change idea.

On the other front, I admit that I haven't given this a lot of mind share, and I imagine future developers won't as well.  This change just feels too risky, because there is nothing stopping anyone from using `STORAGE_BACKING_FOR_CACHE` the way they should expect to be able to use it.  Even the renaming with `_` prefix is a bit odder, and even more so if we plan to use it in all the tests.

If the old implementation had this encapsulated in the switch, why couldn't this do the same with a WaffleSwitch subclass that keeps the legacy functionality?  We can leave comments about where the design came from, but wouldn't it enable the Switch to still be used like a Switch?  This is where I have given this limited mind share, and I am unsure of the implications of my suggestion.

Reply:
> If the old implementation had this encapsulated in the switch, why couldn't this do the same with a WaffleSwitch subclass that keeps the legacy functionality? We can leave comments about where the design came from, but wouldn't it enable the Switch to still be used like a Switch? This is where I have given this limited mind share, and I am unsure of the implications of my suggestion.

Yes, this is possible, but we discussed against it because it requires access to the `_cached_switches` protected property. I can revert to using it, and I think it's actually a better solution.

Reply:
Ok. We could also add a comment to   the protected property that it is being used by subclasses and needs to be treated as part of the public api. And/or, we could remove the _ prefix, but that may have more people trying to dip into the cache.

Reply:
I added the comment to this PR: https://github.com/edx/edx-toggles/pull/107/commits/7d1711f6db9e6018374c9cba0eb17f29c216fde3

Reply:
See the updated PR. Note that I had to squash my commits (sorry about that).